amundsen_application/static/js/components/Tags/TagsList/index.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+
+import ShimmeringTagListLoader from 'components/common/ShimmeringTagListLoader';
+import TagInfo from 'components/Tags/TagInfo';
+import { Tag } from 'interfaces';
+
+import {
+  POPULAR_TAGS_TITLE,
+  CURATED_TAGS_TITLE,
+  BROWSE_MORE_TAGS_TEXT,
+  OTHER_TAGS_TITLE,
+  BROWSE_TAGS_TITLE,
+  BROWSE_PAGE_PATH,
+} from './constants';
+import './styles.scss';
+
+export type TagsListProps = StateFromProps & OwnProps;
+
+export interface StateFromProps {
+  curatedTags: Tag[];
+  popularTags: Tag[];
+  otherTags?: Tag[];
+  isLoading?: boolean;
+}
+
+interface OwnProps {
+  /* determine if we only want curated/popular tags with
+  'Browse more tags' link or the entire tags list */
+  shortTagsList?: boolean;
+}
+
+interface TagsListTitleProps {
+  titleText: string;
+}
+
+const TagsListTitle: React.FC<TagsListTitleProps> = ({
+  titleText,
+}: TagsListTitleProps) => (
+  <h1 className="title-1" id="browse-header">
+    {titleText}
+  </h1>
+);
+
+const TagsListLabel: React.FC<TagsListTitleProps> = ({
+  titleText,
+}: TagsListTitleProps) => (
+  <label className="section-label">
+    <span className="section-title title-2">{titleText}</span>
+  </label>
+);
+
+interface TagsListBlockProps {
+  tags: Tag[];
+}
+
+const TagsListBlock: React.FC<TagsListBlockProps> = ({
+  tags,
+}: TagsListBlockProps) => {
+  return (
+    <div id="tags-list" className="tags-list">
+      {tags.map((tag, index) => (
+        <TagInfo data={tag} compact={false} key={index} />
+      ))}
+    </div>
+  );
+};
+
+const ShortTagsList: React.FC<TagsListProps> = ({
+  curatedTags,
+  popularTags,
+}: TagsListProps) => {
+  return (
+    <div className="short-tag-list">
+      {curatedTags.length === 0 && popularTags.length > 0 && (
+        <TagsListTitle titleText={POPULAR_TAGS_TITLE} />
+      )}
+      {curatedTags.length > 0 && (
+        <TagsListTitle titleText={CURATED_TAGS_TITLE} />
+      )}
+      {curatedTags.length === 0 && popularTags.length > 0 && (
+        <TagsListBlock tags={popularTags} />
+      )}
+      {curatedTags.length > 0 && <TagsListBlock tags={curatedTags} />}
+      <span>
```

Comment:
Curious about this span here, could we avoid it?

Reply:
gone, forgot why I had it in the first place